src/libstore/common-protocol.cc

```diff
@@ -49,13 +49,18 @@ void CommonProto::Serialise<ContentAddress>::write(
 Realisation CommonProto::Serialise<Realisation>::read(const StoreDirConfig & store, CommonProto::ReadConn conn)
 {
     std::string rawInput = readString(conn.from);
-    return Realisation::fromJSON(nlohmann::json::parse(rawInput), "remote-protocol");
+    try {
+        return nlohmann::json::parse(rawInput);
+    } catch (Error & e) {
+        e.addTrace({}, "while parsing the remote protofol");
```

Comment:
Typo, and add the format, if needed

```suggestion
        e.addTrace({}, "while parsing the remote protocol");
```

```suggestion
        e.addTrace({}, "while parsing a realisation object in the remote protocol");
```